gcampus/core/views/parameter.py

```diff
@@ -20,46 +20,46 @@
 from django.http import HttpResponseRedirect, HttpRequest
 from django.shortcuts import get_object_or_404
 from django.urls import reverse
+from django.utils.decorators import method_decorator
 from django.utils.translation import gettext_lazy as _
+from django.utils.translation import gettext
 from django.views.generic.base import TemplateResponseMixin, View
+from django.views.generic.edit import ModelFormMixin, UpdateView
 
 from gcampus.auth.exceptions import UnauthenticatedError, TokenEditPermissionError
 from gcampus.auth.models.token import BaseToken
-from gcampus.core.forms.measurement import ParameterFormSet
+from gcampus.core.decorators import require_permission_edit_measurement
+from gcampus.core.forms.measurement import (
+    ChemicalParameterFormSet,
+    BiologicalParameterFormSet,
+    StructureIndexForm,
+    BaseParameterFormset,
+)
 from gcampus.core.models import Parameter, Measurement
+from gcampus.core.models.index.structure import StructureIndex
 from gcampus.core.views.base import TitleMixin
 
 
-class ParameterFormSetView(TitleMixin, TemplateResponseMixin, View):
-    formset_class = ParameterFormSet
-    template_name = "gcampuscore/forms/parameters.html"
-    next_view_name = "gcampuscore:measurement-detail"
-
-    def get_title(self) -> str:
-        return _("Edit Measurement {pk:d} - Measured Parameters").format(
-            pk=self.instance.pk
-        )
-
-    def get_context_data(self, **kwargs):
-        if "object" not in kwargs:
-            kwargs["object"] = self.instance
-        return super().get_context_data(**kwargs)
+class BaseParameterFormSetView(TitleMixin, TemplateResponseMixin, View):
+    formset_class = BaseParameterFormset
 
     def __init__(self, **kwargs):
-        super(ParameterFormSetView, self).__init__(**kwargs)
+        super(BaseParameterFormSetView, self).__init__(**kwargs)
         self.instance: Optional[Parameter] = None
```

Comment:
This mistake has been present since before your PR, but I am fairly certain this has to be of type `Optional[Measurement]`.